acceptable/__main__.py

```diff
@@ -0,0 +1,118 @@
+        raw_args = sys.argv[1:]
+    if parser_cls is None:
+        parser_cls = argparse.ArgumentParser
+    if stdin is None:
+        stdin = sys.stdin
+
+    parser = parser_cls(
+        description='Tool for working with acceptable metadata',
+    )
+    subparser = parser.add_subparsers(dest='cmd')
+    subparser.required = True
+
+    metadata_parser = subparser.add_parser(
+        'metadata', help='Scan files and print extracted metadata in json')
+    metadata_parser.add_argument('files', nargs='+')
+    metadata_parser.set_defaults(func=metadata_cmd)
+
+    render_parser = subparser.add_parser(
+        'render', help='Render markdown documentation for api metadata'
+    )
+    render_parser.add_argument(
+        'metadata',
+        nargs='?',
+        type=argparse.FileType('r'),
+        default=stdin,
+    )
+    render_parser.add_argument(
+        '--name', '-n', required=True, help='Name of service')
+    render_parser.add_argument(
+        '--dir', '-d', default='docs', help='output directory')
+    render_parser.set_defaults(func=render_cmd)
+
+    return parser.parse_args(raw_args)
+
+
+def metadata_cmd(cli_args):
+    metadata = scan_metadata(cli_args.files)
+    print(json.dumps(metadata, indent=2, sort_keys=True))
+    return 0
+
+
+def scan_metadata(files):
+    # This uses AST parsing for now. In future, we might just import code.
+    schemas = []
+    metadata = {}
+    for path in files:
+        schemas.extend(extract_schemas_from_file(path))
+    for schema in schemas:
+        metadata[schema.view_name] = {
+            'api_name': schema.view_name,
+            'version': int(float(schema.version)),  # convert any '1.0' strings
+            'methods': schema.methods,
+            'url': schema.url,
+            'request_schema': schema.input_schema,
+            'response_schema': schema.output_schema,
+            'doc': schema.doc,
+        }
+    return metadata
+
+
+def render_cmd(cli_args):
+    root_dir = Path(cli_args.dir)
+    root_dir.joinpath('en').mkdir(parents=True, exist_ok=True)
+    metadata = json.load(cli_args.metadata)
+
+    for path, content in render_markdown(metadata, cli_args.name):
+        root_dir.joinpath(path).write_text(content)
+
+
+def render_markdown(metadata, name):
+    docs_metadata = {
+        'navigation': [{'title': 'Index', 'location': 'index.md'}]
+    }
+    pages = []
+    en = Path('en')
+    page = env.get_template('api_page.md.j2')
+    index = env.get_template('index.md.j2')
+
+    for api_name, api in metadata.items():
+        page_file = '{}.md'.format(api_name)
+        pages.append({'title': api_name, 'location': page_file})
+        yield en / page_file, page.render(name=api_name, **api)
+
+    docs_metadata['navigation'].extend(
+        sorted(pages, key=lambda k: k['title']))
+
+    yield en / 'index.md', index.render(service_name=name)
+    yield en / 'metadata.yaml', yaml.safe_dump(docs_metadata)
+    yield Path('metadata.yaml'), yaml.safe_dump(
```

Comment:
It's not clear to me why we write yaml files. The documentation in README mentions .json files, but no yaml files.

Why do weuse both json and yaml? Why do we render yaml files here, when this command is ostensibly to build human-readable output? Is this a requirement of documentation-builder? If so, perhaps that could be made more obvious in the code itself.